### ISSUE TITLE
TCN with GMM return_params=true failing

### ISSUE THREAD

**Issue by dluuo** [Nixtla/neuralforecast#430]
![image](https://user-images.githubusercontent.com/60141418/216683060-d9e91d11-bfb2-41d4-b3a8-b65b35edb548.png)
When TCN is called with loss = GMM(...,return_params=true,...), i get the following error:
![image](https://user-images.githubusercontent.com/60141418/216682904-46659296-4d4f-47de-8bb0-07eea8ebff9e.png)
This issue is caused by an incorrect reshaping in the "if.self.loss.return_params" section:
![image](https://user-images.githubusercontent.com/60141418/216683017-f8f5d716-a1f7-48c4-87c0-ba0a3b772c0c.png)

**Comment by dluuo** [comment]
Fix is here: #431
